src/iuse.cpp

```diff
@@ -8293,3 +8293,37 @@ int iuse::saw_barrel( player *p, item *, bool, const tripoint& )
 
     return 0;
 }
+
+int iuse::washclothes( player *p, item *it, bool, const tripoint& )
+{
+    if( it->charges < it->type->charges_to_use() ) {
+        p->add_msg_if_player( _( "You need a soap to use this." ) );
+        return 0;
+    }
+    
+    const inventory &crafting_inv = p->crafting_inventory();
+    if( !crafting_inv.has_amount( "water", 40 ) && !crafting_inv.has_amount( "water_clean", 40 ) ) {
+        p->add_msg_if_player( _( "You need a large amount of fresh water to use this." ) );
+        return 0;
+    }
+    
+    int pos = g->inv_for_filter( _( "Wash what?" ), []( const item & itm ) {
+        return itm.has_flag( "FILTHY" ); 
+    } );
+    item &mod = p->i_at( pos );
+    if( pos == INT_MIN ) {
+        p->add_msg_if_player( m_info, _( "Never mind." ) );
+        return 0;
+    }
+
+    std::vector<item_comp> comps;
+    comps.push_back( item_comp( "water", 40 ) );
+    comps.push_back( item_comp( "water_clean", 40 ) );
+    p->consume_items( comps );
```

Comment:
I mean after that, when he has to choose a source of water (if ambiguous). The menu usually allows cancel.